com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/dao/APIDAO.java

```diff
@@ -882,4 +882,56 @@ public static boolean inAPIQuotaDateRange(String serviceProvider,String applicat
 		return inQuotaDateRange;
 	}
 
+	public boolean isBlacklistedorWhitelistedNumber(String msisdn, String apiId,
+													String appId, String subscriberId,
+													String action) throws Exception {
+
+		Connection connection = null;
+		PreparedStatement preparedStatement = null;
+		ResultSet resultSet = null;
+		boolean isListed = false;
+
+		try {
+			StringBuilder queryBuilder = new StringBuilder();
+
+			queryBuilder.append("select exists ");
+			queryBuilder.append(" ( ");
+			queryBuilder.append("select 1 FROM ");
+			queryBuilder.append(DatabaseTables.BLACKLIST_WHITELIST_MSISDN + "  ");
+			queryBuilder.append(" where ");
+			queryBuilder.append(" (API_ID = '%' OR API_ID = ? )");
+			queryBuilder.append(" AND MSISDN = ? ");
+			queryBuilder.append(" AND (APP_ID = '%' OR APP_ID = ? )");
+			queryBuilder.append(" AND SERVICE_PROVIDER = ? ");
+			queryBuilder.append(" AND ACTION = ? ");
+			queryBuilder.append(" ) ");
+			queryBuilder.append(" AS result ");
```

Comment:
No need of using a StringBuilder here. Define this as a constant and refer it in the code. It'll be evaluated at the compile time.